guard/src/commands/external.rs

```diff
@@ -0,0 +1,3 @@
+pub trait Pull {
```

Comment:
Can we instead create a trait for a generic source, call it `ExternalSource` and make it contain all the methods that a typical source should have? Think from perspective of all new sources that we might add and what will be the common denominator.